zrml/swaps/src/fixed.rs

```diff
@@ -180,14 +232,166 @@ mod tests {
+            (10000000000, 7000000000, 10000000000),
+            (10000000000, 8000000000, 10000000000),
+            (10000000000, 9000000000, 10000000000),
+            (10000000000, 10000000000, 10000000000),
+            (12500000000, 0, 10000000000),
+            (12500000000, 1000000000, 10225651825),
+            (12500000000, 2000000000, 10456395525),
+            (12500000000, 3000000000, 10692345999),
+            (12500000000, 4000000000, 10933620739),
+            (12500000000, 5000000000, 11180339887),
+            (12500000000, 6000000000, 11432626298),
+            (12500000000, 7000000000, 11690605597),
+            (12500000000, 8000000000, 11954406247),
+            (12500000000, 9000000000, 12224159606),
+            (12500000000, 10000000000, 12500000000),
+            (15000000000, 0, 10000000000),
+            (15000000000, 1000000000, 10413797439),
+            (15000000000, 2000000000, 10844717711),
+            (15000000000, 3000000000, 11293469354),
+            (15000000000, 4000000000, 11760790225),
+            (15000000000, 5000000000, 12247448713),
+            (15000000000, 6000000000, 12754245006),
+            (15000000000, 7000000000, 13282012399),
+            (15000000000, 8000000000, 13831618672),
+            (15000000000, 9000000000, 14403967511),
+            (15000000000, 10000000000, 15000000000),
+            (17500000000, 0, 10000000000),
+            (17500000000, 1000000000, 10575570503),
+            (17500000000, 2000000000, 11184269147),
+            (17500000000, 3000000000, 11828002689),
+            (17500000000, 4000000000, 12508787635),
+            (17500000000, 5000000000, 13228756555),
+            (17500000000, 6000000000, 13990164762),
+            (17500000000, 7000000000, 14795397379),
+            (17500000000, 8000000000, 15646976811),
+            (17500000000, 9000000000, 16547570643),
+            (17500000000, 10000000000, 17500000000),
+        ];
+        for (base, exp, expected) in test_vector.iter() {
+            let result = bpow_approx(*base, *exp).unwrap();
+            let diff = if result > *expected { result - *expected } else { *expected - result };
+            assert_le!(diff, precision);
+        }
+    }
+
+    #[test]
+    fn bpow_approx_returns_error_when_parameters_are_outside_of_specified_limits() {
+        let test_vector: Vec<(u128, u128, DispatchError)> = vec![
+            (BASE, BASE + 1, DispatchError::Other("[bpow_approx]: expected exp <= BASE")),
+            (
+                BASE / 10,
+                1 * BASE / 2,
+                DispatchError::Other("[bpow_approx]: expected base >= BASE / 4"),
+            ),
+            (
+                2 * BASE - BASE / 10,
+                1 * BASE / 2,
+                DispatchError::Other("[bpow_approx]: expected base <= 7 * BASE / 4"),
+            ),
+        ];
+        for (base, exp, err) in test_vector.iter() {
+            assert_eq!(bpow_approx(*base, *exp).unwrap_err(), *err);
```

Comment:
The `frame_support` crate offers a macro for that specific case: `assert_err!(result, excepted_error)`
```suggestion
            frame_support::assert_err!(bpow_approx(*base, *exp), *err);
```